src/App.jsx

```diff
@@ -14,33 +17,118 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getSortGoods(goods, { sortField, reversed }) {
+  let prepGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    prepGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reversed) {
+    prepGoods = prepGoods.reverse();
+  }
+
+  return prepGoods;
+}
+
+export const App = () => {
+  const [resetIsVisibe, setResetIsVisibe] = useState(false);
+  const [reversed, setReversed] = useState(false);
+  const [sortField, setSortField] = useState('');
+  const visibleGoods = getSortGoods(goodsFromServer, { sortField, reversed });
+
+  const reverse = () => {
+    setReversed(!reversed);
+
+    if (!reversed) {
+      setResetIsVisibe(true);
+    } else if (reversed && sortField !== '') {
+      setResetIsVisibe(true);
+    } else if (reversed) {
+      setResetIsVisibe(false);
+    }
+  };
```

Comment:
```suggestion
  const reverse = () => setReversed(prevIsReversed =>  !prevIsReversed);
```
all what should to do reverse button it is set isReversed state to opposite